File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.go
import "sort"

func canMakeArithmeticProgression(arr []int) bool {
    sort.Ints(arr)
    var store int = arr[1] - arr[0]
    
    var j int
    var i int = 0
    for j = 1; j < len(arr); j += 1 {
        curr := arr[j] - arr[i]
        if store != curr {
            return false
        }
    
        i += 1
    }
    return true
}

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        i = 0
        j = 1
        store = arr[j] -arr[i]
        while j < len(arr):
            curr = arr[j] - arr[i]
            if curr != store:
                return False
            j += 1
            i += 1
        return True
            